### ISSUE TITLE
TelemetryClient.TrackTrace creates a new operation_parentId

### ISSUE THREAD

**Issue by GeeWee** [microsoft/ApplicationInsights-dotnet#1339]
Consider the following example.

```csharp
var activity = new Activity("generateEvents");
using (var operationHolder = _telemetry.StartOperation<RequestTelemetry>(activity))
{                
_telemetry.TrackTrace($"Trace. Current activity {activity.TraceId}, parent: {activity.Parent.SpanId}. OperationId {telemetryOperation.Telemetry.Context.Operation.Id}, parentId={telemetryOperation.Telemetry.Context.Operation.ParentId}");
}
_telemetry.Flush();
```

Results in this message in Application Insights:
```
Trace. Current activity 3a199ac57afe9548800bd166378be4bc, parent: e76b75760867844c. OperationId 3a199ac57afe9548800bd166378be4bc, parentId=|3a199ac57afe9548800bd166378be4bc.e76b75760867844c.
```

And this operationId: `3a199ac57afe9548800bd166378be4bc` in the `operation_Id` column, which is fine. That matches the operationId reported from the `operationHolder`.

The operation_parentId however, does not match, and returns
`|3a199ac57afe9548800bd166378be4bc.d90e9f42fff8144b.`, where I would expect it to return the same parentId as the operationHolder, namely `|3a199ac57afe9548800bd166378be4bc.e76b75760867844c.`

Is this a bug, or am I misunderstanding something with regards to how these ID's work?

**Comment by cijothomas** [comment]
var activity = new Activity("generateEvents");
This activity is not started - can you call activity.Start, and see if it matches your expectations? ( i havent fully understood your end goal, sorry so this may not help :( )

**Comment by GeeWee** [comment]
I don't think I'm meant to start the activity, considering the method documentation says.
```When Activity instance is passed to StartOperation, it is expected that Activity has ParentId (if it was provided by upstream service), but has not been started yet.
It may also have additional Tags and Baggage to augment telemetry.
```
I have tried to both start the activity first, via `activity.Start()`, and use the `startOperation` that just takes in an `operationName` parameter, and both result in the same behaviour.

A little more on what I'm trying to do - this issue is related to #1338, where I'm trying to create an integration with Serilog, which properly includes `operation_id` and `operation_parentid`.
So basically I'd like to be able to instrument my logs with the `operation_parentId` - and have it match the one from the `TelemetryClient.TrackTrace()` - but I'm unable to do so, as the TelemetryClient doesn't seem to use the `parentId` of the activity started via the `startOperation` call.

Perhaps I'm going about this all wrong though.

**Comment by lmolkova** [comment]
Activity is created to track the request telemetry (with StartOperation). The trace telemetry, tracked in scope of request, becomes a child of the request telemetry, thus trace's parent matcher request's id (and contains Activity spanId).

As I mentioned in #1338, you do not need to change ids, just add `OperationCorrelationTelemetryInitializer` to your configuration and it will take care of all ids.

**Comment by GeeWee** [comment]
Yes, this is due to a missing `OperationCorrelationTelemetryInitializer` - sorry about that. Closing.
